v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -57,7 +62,196 @@ async function getTasksAndHreEnvironment(
   };
 }
 
+async function getTasksAndSubtaskResults(
+  configFileName: string = "hardhat.config.ts",
+): Promise<boolean[]> {
+  // To ensure that one or more tasks have been executed, each task will modify an array of boolean values, initially set to false.
+  // This function imports that array, allowing the tests to verify if the tasks have been executed.
+  // If a boolean flag is true, it indicates that the corresponding task (or a specific part of it) has been executed.
+  // The array is set in the hardhat.config.ts file of the fixture project.
+  return (
+    await import(pathToFileURL(`${process.cwd()}/${configFileName}`).toString())
+  ).results;
+}
+
 describe("main", function () {
+  describe("main", function () {
+    afterEach(function () {
+      resetHardhatRuntimeEnvironmentSingleton();
+      mock.reset();
+    });
+
+    describe("version", function () {
+      useFixtureProject("cli/parsing/base-project");
+
+      // TODO: as soon as the 'version task' is done, this test should be updated
+      it("should print the version and instantly return", async function () {
+        const m = mock.method(console, "log", () => {});
+        const spy = mock.fn(importUserConfig);
+
+        const command = "npx hardhat --version";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        assert.equal(m.mock.calls.length, 1);
+        assert.equal(m.mock.calls[0].arguments[0], "3.0.0");
```

Comment:
can't we get the hardhat version from the `package.json`? Otherwise, every time hardhat is published, this test will fail.

Reply:
Yes, this is a TODO, it is already implemented in this PR: https://github.com/NomicFoundation/hardhat/pull/5384